### PR TITLE
Generate the REST nonce suspending WC filters

### PR DESCRIPTION
**Pull request by lucatume**
Issue: https://moderntribe.atlassian.net/browse/TEC-3579

[Screencast](https://drive.google.com/open?id=1CUDUUP10O2N1n9-bm47NTcdpuaE-FXZX&authuser=luca%40tri.be&usp=drive_fs)

This PR fixes the issue that was caused by the following combination:

1. the user is not logged in
2. WooCommerce will filter the `nonce_user_logged_out` filter to set the user id to a hash used for cart tracking
3. the user id is part of the salts used to generated the nonces, including the views nonce, during the PHP initial state render
4. WooCommerc will not filter the `nonce_user_logged_out` filter during AJAx requests
5. during the AJAX requests fired by the Views to fetch the new HTML the nonce provided (salted with the hash user id) will not match the one generated without applying hashes (point 4: WC is not filtering that during AJAX requests)

The fix leverages the `tribe_without_filters` function, introduced in [the common PR](https://github.com/moderntribe/tribe-common/pull/1436) to generate the view REST nonce correctly, suspending filtering that might be applied to it.

### CODE REVIEW

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 3 error(s)
:warning: 1 warning(s)

File: src/Tribe/Views/V2/View.php [hunk 1]
```diff
@@ -1393,7 +1393,16 @@ protected function setup_template_vars() {
 			? Dates::build_date_object( $event_date )->format( Dates::DBDATEFORMAT )
 			: false;
 
-		$template_vars = [
+		/*
+		 * Some plugins, like WooCommerce, will modify the UID of logged out users; avoid that filtering here.
+		 *
+		 * @see TEC-3579
+		 */
+		$rest_nonce = tribe_without_filters( [ 'nonce_user_logged_out' ], static function () {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1401]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: f2fad6b1238e0ade376c833916899bee</sub>

File: src/Tribe/Views/V2/View.php [hunk 2]
```diff
@@ -1393,7 +1393,16 @@ protected function setup_template_vars() {
...
+			return wp_create_nonce( 'wp_rest' );
+		} );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1403]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 9a81805eb2615ddd4d5215a5db136b01</sub>

File: src/Tribe/Views/V2/View.php [hunk 3]
```diff
@@ -1393,7 +1393,16 @@ protected function setup_template_vars() {
...
+
+		$template_vars   = [
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1405]: :warning: Equals sign not aligned correctly; expected 1 space but found 3 spaces

<sub>hash: 2cae0f95f1223b4d001dcaf9a7fdde0f</sub>

Comment (Camwyn) [reply src/Tribe/Views/V2/View.php:1405]:
```suggestion
		$template_vars = [
```

File: src/Tribe/Views/V2/View.php (hunk 2 above)
Comment (Camwyn) [inline src/Tribe/Views/V2/View.php:1403]:
```suggestion
		$rest_nonce = tribe_without_filters(
			[ 'nonce_user_logged_out' ],
			static function () {
				return wp_create_nonce( 'wp_rest' );
			}
		);
```

**Review by Camwyn — DISMISSED** [review]
Just suggested fixes to quiet the bot - otherwise looks great.

**Review by Camwyn — DISMISSED** [review]

**Review by paulmskim — APPROVED** [review]

**Pull request merged**